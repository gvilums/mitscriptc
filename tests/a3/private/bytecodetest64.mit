x = (5 == "5");
if (x) {
    print("x is true");
} else {
    print("x is false");
}
print(5 == intcast("5"));

if (!x) {
    print("x is still false");
}

while (x) {
    print("Loop!");
}

i = 0;
while (!x) {
    print("Loop " + i);
    x = i >= 5;
    i = i + 1;
}

arr = {};
i = 0;
while (i < 4) {
    arr[i] = "value " + i;
    i = i + 1;
}
print(arr);
arr["1"] = "foo";
print(arr);
print(arr.cat);
print(arr[2]);
print(arr[5]);

rec = { foo: "bar"; cat: "dog"; };
print(rec);
print(rec.cat);
rec.cat = rec.cat + " cat";
print(rec.cat);
print(1 + 2 * 3);
print("cat" + 1 + 2 * 3 + None + print + true + false);
print(input);
x = (1 + (1 + (2 * 3)) + 5 / 6 + 4 / 2 - (((9 * 10))));
print(x);
print(5 > 5);
print(5 > 4);
print("foo" == "bar");
print(print == input);
print("hi" + print == "hi" + input);
rec = { foo: "bar"; cat: "dog"; };
rec0 = { foo: "bar"; cat: "dog"; };
print(rec == rec0);
rec2 = rec;
print(rec == rec2);
rec.foo = "foobar";
print(rec);
print(rec2);
print(rec==rec2);
x = 1;
y = x;
x = 2;
print(y);
print(1+2==3);
print("foo" + "bar" == "foobar");
print("5" == 5);
print("5" + 5 == "5" + "5");
print(None);
print(true & false);
print(true & true);
print(false & false);
print(false | true);
print(true | true);
print(false | false);
print(5 - 8 == -3);

global k;

print(1 + "2");
print("1" + 2);
print(1 + "2" == "1" + 2);

return 0;

print("Done!");


