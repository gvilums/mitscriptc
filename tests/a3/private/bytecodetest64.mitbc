function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		}
	],
	constants = [5, "5", "x is true", "x is false", "x is still false", "Loop!", 0, "Loop ", 1, 4, "value ", "1", "foo", 2, "bar", "dog", " cat", 3, "cat", None, true, false, 6, 9, 10, "hi", "foobar", 8, "2", "Done!"],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [k, x, i, arr, rec, bar, rec0, rec2, y, print, intcast, cat, foo, input],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	9
		load_func	1
		alloc_closure	0
		store_global	13
		load_func	2
		alloc_closure	0
		store_global	10
		load_const	0
		load_const	1
		eq
		store_global	1
		load_global	1
		not
		if	6
		load_global	9
		load_const	2
		call	1
		pop
		goto	5
		load_global	9
		load_const	3
		call	1
		pop
		load_global	9
		load_const	0
		load_global	10
		load_const	1
		call	1
		eq
		call	1
		pop
		load_global	1
		not
		not
		if	5
		load_global	9
		load_const	4
		call	1
		pop
		load_global	1
		not
		if	6
		load_global	9
		load_const	5
		call	1
		pop
		goto	-7
		load_const	6
		store_global	2
		load_global	1
		not
		not
		if	16
		load_global	9
		load_const	7
		load_global	2
		add
		call	1
		pop
		load_global	2
		load_const	0
		geq
		store_global	1
		load_global	2
		load_const	8
		add
		store_global	2
		goto	-18
		alloc_record
		store_global	3
		load_const	6
		store_global	2
		load_global	2
		load_const	9
		swap
		gt
		not
		if	12
		load_global	3
		load_global	2
		load_const	10
		load_global	2
		add
		index_store
		load_global	2
		load_const	8
		add
		store_global	2
		goto	-16
		load_global	9
		load_global	3
		call	1
		pop
		load_global	3
		load_const	11
		load_const	12
		index_store
		load_global	9
		load_global	3
		call	1
		pop
		load_global	9
		load_global	3
		field_load	11
		call	1
		pop
		load_global	9
		load_global	3
		load_const	13
		index_load
		call	1
		pop
		load_global	9
		load_global	3
		load_const	0
		index_load
		call	1
		pop
		alloc_record
		dup
		dup
		load_const	14
		field_store	12
		load_const	15
		field_store	11
		store_global	4
		load_global	9
		load_global	4
		call	1
		pop
		load_global	9
		load_global	4
		field_load	11
		call	1
		pop
		load_global	4
		load_global	4
		field_load	11
		load_const	16
		add
		field_store	11
		load_global	9
		load_global	4
		field_load	11
		call	1
		pop
		load_global	9
		load_const	8
		load_const	13
		load_const	17
		mul
		add
		call	1
		pop
		load_global	9
		load_const	18
		load_const	8
		add
		load_const	13
		load_const	17
		mul
		add
		load_const	19
		add
		load_global	9
		add
		load_const	20
		add
		load_const	21
		add
		call	1
		pop
		load_global	9
		load_global	13
		call	1
		pop
		load_const	8
		load_const	8
		load_const	13
		load_const	17
		mul
		add
		add
		load_const	0
		load_const	22
		div
		add
		load_const	9
		load_const	13
		div
		add
		load_const	23
		load_const	24
		mul
		sub
		store_global	1
		load_global	9
		load_global	1
		call	1
		pop
		load_global	9
		load_const	0
		load_const	0
		gt
		call	1
		pop
		load_global	9
		load_const	0
		load_const	9
		gt
		call	1
		pop
		load_global	9
		load_const	12
		load_const	14
		eq
		call	1
		pop
		load_global	9
		load_global	9
		load_global	13
		eq
		call	1
		pop
		load_global	9
		load_const	25
		load_global	9
		add
		load_const	25
		load_global	13
		add
		eq
		call	1
		pop
		load_global	9
		load_global	9
		load_global	9
		eq
		call	1
		pop
		load_global	9
		store_global	5
		load_global	9
		load_global	5
		load_global	9
		eq
		call	1
		pop
		alloc_record
		dup
		dup
		load_const	14
		field_store	12
		load_const	15
		field_store	11
		store_global	4
		alloc_record
		dup
		dup
		load_const	14
		field_store	12
		load_const	15
		field_store	11
		store_global	6
		load_global	9
		load_global	4
		load_global	6
		eq
		call	1
		pop
		load_global	4
		store_global	7
		load_global	9
		load_global	4
		load_global	7
		eq
		call	1
		pop
		load_global	4
		load_const	26
		field_store	12
		load_global	9
		load_global	4
		call	1
		pop
		load_global	9
		load_global	7
		call	1
		pop
		load_global	9
		load_global	4
		load_global	7
		eq
		call	1
		pop
		load_const	8
		store_global	1
		load_global	1
		store_global	8
		load_const	13
		store_global	1
		load_global	9
		load_global	8
		call	1
		pop
		load_global	9
		load_const	8
		load_const	13
		add
		load_const	17
		eq
		call	1
		pop
		load_global	9
		load_const	12
		load_const	14
		add
		load_const	26
		eq
		call	1
		pop
		load_global	9
		load_const	1
		load_const	0
		eq
		call	1
		pop
		load_global	9
		load_const	1
		load_const	0
		add
		load_const	1
		load_const	1
		add
		eq
		call	1
		pop
		load_global	9
		load_const	19
		call	1
		pop
		load_global	9
		load_const	20
		load_const	21
		and
		call	1
		pop
		load_global	9
		load_const	20
		load_const	20
		and
		call	1
		pop
		load_global	9
		load_const	21
		load_const	21
		and
		call	1
		pop
		load_global	9
		load_const	21
		load_const	20
		or
		call	1
		pop
		load_global	9
		load_const	20
		load_const	20
		or
		call	1
		pop
		load_global	9
		load_const	21
		load_const	21
		or
		call	1
		pop
		load_global	9
		load_const	0
		load_const	27
		sub
		load_const	17
		neg
		eq
		call	1
		pop
		load_global	9
		load_const	8
		load_const	28
		add
		call	1
		pop
		load_global	9
		load_const	11
		load_const	13
		add
		call	1
		pop
		load_global	9
		load_const	8
		load_const	28
		add
		load_const	11
		load_const	13
		add
		eq
		call	1
		pop
		load_const	6
		return
		load_global	9
		load_const	29
		call	1
		pop
	]
}