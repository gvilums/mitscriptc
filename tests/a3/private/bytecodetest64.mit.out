x is false
true
x is still false
Loop 0
Loop 1
Loop 2
Loop 3
Loop 4
Loop 5
{0:value 0 1:value 1 2:value 2 3:value 3 }
{0:value 0 1:foo 2:value 2 3:value 3 }
None
value 2
None
{cat:dog foo:bar }
dog
dog cat
7
cat16NoneFUNCTIONtruefalse
FUNCTION
-80
false
true
false
false
true
false
true
{cat:dog foo:foobar }
{cat:dog foo:foobar }
true
1
true
true
false
true
None
false
true
false
true
true
false
true
12
12
true
